src/DbPostgreSQLAdapter/Table.cpp

```diff
@@ -0,0 +1,660 @@
+#include "stdafx.h"
+#include "Table.h"
+
+#include "Database.h"
+#include "Field.h"
+#include "FieldValue.h"
+#include "PrimaryKey.h"
+#include "PrimaryKeyValue.h"
+#include "TableRecord.h"
+
+#include <DbAdapterInterface/IRecordSet.h>
+#include <DbAdapterInterface/IRecord.h>
+
+namespace {
+	std::string dateTimeToISOString(const std::chrono::system_clock::time_point& dateTime)
+	{
+		if (dateTime != std::chrono::system_clock::time_point{})
+		{
+			return std::format("{:%F %T%z}", dateTime);
+		}
+
+		return "";
+	}
+
+	std::string getSQLValue(const systelab::db::IFieldValue& fieldValue, bool forComparison, bool forAssignment)
+	{
+		std::ostringstream fieldValueStream;
+		if (fieldValue.isNull())
+		{
+			if (forComparison)
+			{
+				fieldValueStream << " IS ";
+			}
+			else if (forAssignment)
+			{
+				fieldValueStream << " = ";
+			}
+
+			fieldValueStream << "NULL";
+		}
+		else
+		{
+			if (forComparison || forAssignment)
+			{
+				fieldValueStream << " = ";
+			}
+
+			const systelab::db::FieldTypes fieldType = fieldValue.getField().getType();
+			switch (fieldType)
+			{
+			case systelab::db::BOOLEAN:
+				fieldValueStream << (fieldValue.getBooleanValue() ? "True" : "False");
+				break;
+			case systelab::db::INT:
+				fieldValueStream << fieldValue.getIntValue();
+				break;
+			case systelab::db::DOUBLE:
+				fieldValueStream << std::fixed << std::setprecision(10) << fieldValue.getDoubleValue();
```

Comment:
10 seems like a lot, generally 6 is good enough for science

Reply:
This should b removed, its a leftover